### PR TITLE
[Magiclysm] Adding new magic plants

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary

Mods "Adding new magic plants"

#### Purpose of change
Reopened #37103 with some changes

- [x] fix errors
- [x] make a rare map for some of them
- [x] make some special effects for plants
- [x] desing their using in alchemy
- [x] spawn plants in the map

### CODE REVIEW

**Comment by GuardianDll** [comment]
btw am i need to add this herb names to spell checker ignore list?

**Comment by anothersimulacrum** [comment]
I don't believe furniture and terrain can appear as plurals in-game, which is why you are seeing those problems.

**Comment by GuardianDll** [comment]
DAAAAAMN
its a furniture, when it must be `COMESTIBLE`
🤦

**Comment by GuardianDll** [comment]
Accidentally turn off the draft mode, sorry

File: data/mods/Magiclysm/items/herbs_and_flowers.json [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    "comestible_type": "FOOD",
+    "description": "A handful of moonflowers glitter like tiny stars and faintly smell of morning dew.  Properly harvested, they will never wilt or wither.",
+    "volume": "250 ml",
+    "vitamins": [ [ "vitC", 5 ], [ "iron", 3 ], [ "calcium", 2 ] ]
+  },
+  {
+    "type": "COMESTIBLE",
+    "id": "kämpuch",
+    "copy-from": "platonic_flower",
+    "name": { "str": "kämpuch", "str_pl": "kämpuch" },
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/herbs_and_flowers.json:33]: Plural is probably +es

File: data/mods/Magiclysm/items/herbs_and_flowers.json [hunk 2]
```diff
@@ -0,0 +1,156 @@
+[
+  {
+    "type": "COMESTIBLE",
+    "id": "platonic_flower",
+    "name": { "str": "philosopher's flower", "str_pl": "philosopher's flowers" },
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/herbs_and_flowers.json:5]: No need for explicit plural here

File: data/mods/Magiclysm/items/herbs_and_flowers.json [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    "calories": 800,
+    "comestible_type": "FOOD",
+    "looks_like": "wheat",
+    "description": "Stalks cut from a golden reed.  They have a strong, spiced aroma."
+  },
+  {
+    "type": "COMESTIBLE",
+    "id": "wood_barnacle",
+    "copy-from": "platonic_flower",
+    "name": { "str": "wood barnacle", "str_pl": "wood barnacles" },
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/herbs_and_flowers.json:81]: No need for explicit plural

File: data/mods/Magiclysm/items/herbs_and_flowers.json [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    "calories": 800,
+    "comestible_type": "FOOD",
+    "looks_like": "egg",
+    "description": "Heavy wooden shells surrounding a diminutive white fruit."
+  },
+  {
+    "type": "COMESTIBLE",
+    "id": "dragon_nail",
+    "copy-from": "platonic_flower",
+    "name": { "str": "dragon nails", "str_pl": "dragonails" },
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/herbs_and_flowers.json:92]: Looks like a typo

File: data/mods/Magiclysm/items/herbs_and_flowers.json [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    "calories": 800,
+    "comestible_type": "FOOD",
+    "looks_like": "egg",
+    "description": "Knife sharp thorns from a dragonail bush.  Eating them raw would be a dangerous challenge."
+  },
+  {
+    "type": "COMESTIBLE",
+    "id": "moly_tuber",
+    "copy-from": "platonic_flower",
+    "name": { "str": "moly tuber", "str_pl": "moly tubers" },
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/herbs_and_flowers.json:103]: No need for explicit plural

File: data/mods/Magiclysm/items/herbs_and_flowers.json [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    "symbol": "%",
+    "calories": 70,
+    "comestible_type": "FOOD",
+    "description": "The black root of a moly plant, extracted from the ground through great effort.  Rumored to grant protection to those who consume it."
+  },
+  {
+    "type": "COMESTIBLE",
+    "id": "fern_flower",
+    "copy-from": "platonic_flower",
+    "name": { "str": "moly tuber", "str_pl": "moly tubers" },
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/herbs_and_flowers.json:114]: This should probably have a different name

File: data/mods/Magiclysm/items/herbs_and_flowers.json [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    "symbol": "%",
+    "calories": 70,
+    "comestible_type": "FOOD",
+    "description": "A small flowwer cut from a fern, it has a faint honeyed aroma.  Believed to bring good luck to whoever finds it."
+  },
+  {
+    "type": "COMESTIBLE",
+    "id": "funnelcap_chunk",
+    "copy-from": "platonic_flower",
+    "name": { "str": "funnelcap chunk", "str_pl": "funnelcap chunks" },
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/herbs_and_flowers.json:125]: No need for explicit plural

File: data/mods/Magiclysm/items/herbs_and_flowers.json [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+    "symbol": "%",
+    "calories": 70,
+    "comestible_type": "FOOD",
+    "description": "Large chunks of funnelcap meat, still leaking fine amounts of clear sap."
+  },
+  {
+    "type": "COMESTIBLE",
+    "id": "sojourners_flower",
+    "copy-from": "platonic_flower",
+    "name": { "str": "sojourner's flower", "str_pl": "sojourner's flowers" },
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/herbs_and_flowers.json:136]: No need for explicit plural

File: data/mods/Magiclysm/items/herbs_and_flowers.json [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+    "calories": 10,
+    "comestible_type": "FOOD",
+    "description": "A red puffball flower cut from a sojourner's palm.  When wind wafts between its petals, you can hear the sounds of faraway places.",
+    "volume": "10 ml"
+  },
+  {
+    "type": "COMESTIBLE",
+    "id": "whiskflower",
+    "copy-from": "platonic_flower",
+    "name": { "str": "whiskflower", "str_pl": "whiskflowers" },
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/herbs_and_flowers.json:148]: No need for explicit plural

**Review by esotericist — CHANGES_REQUESTED** [review]
this is a good start, and they sound interesting, but as discussed on discord i think before we can move forward we're going to want to fill out those checkboxes from @John-Candlebury 's original pr:

* there need to be basic effects for consuming them (or at the very least solid concepts for what they *might* do, if their basic consumption wouldn't be useful/reasonable compared to their future alchemy implementation uses)
* they need to properly be distributed in worldgen so that they can be found in places that make sense to look for them.

also note that one option if you can only think of a subset of things to do with them is to make a new PR with a subset of the changes, that just implement a smaller number of plants with effects and spawn locations. if you need help implementing those, i'm sure someone in #dev-help can give you some pointers once you have some specific questions.

**Comment by GuardianDll** [comment]
Left it here so i wont miss it in the future
IMG_20220419_074422

File: data/mods/Magiclysm/furniture_and_terrain/furniture_plants.json [hunk 10]
```diff
@@ -0,0 +1,211 @@
+[
+  {
+    "type": "furniture",
+    "id": "f_platonic_flower",
+    "name": "Moonflower Bush",
```
Comment (Night-Pryanik) [inline data/mods/Magiclysm/furniture_and_terrain/furniture_plants.json:5]: I suppose this flower should be named somehow different?

File: data/mods/Magiclysm/furniture_and_terrain/furniture_plants.json [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+    "required_str": -1,
+    "flags": [ "TRANSPARENT", "TINY", "FLAMMABLE_ASH", "NOCOLLIDE", "ORGANIC" ],
+    "examine_action": "harvest_furn",
+    "harvest_by_season": [ { "seasons": [ "spring", "summer", "autumn" ], "id": "kampuch_harv" } ],
+    "bash": { "str_min": 2, "str_max": 6, "sound": "crunch.", "sound_fail": "whish." }
+  },
+  {
+    "type": "furniture",
+    "id": "f_olieribos",
+    "name": "t_underbrush",
```
Comment (Night-Pryanik) [inline data/mods/Magiclysm/furniture_and_terrain/furniture_plants.json:50]: Name, not id.

File: data/mods/Magiclysm/furniture_and_terrain/furniture_plants.json [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+    "required_str": -1,
+    "flags": [ "TRANSPARENT", "TINY", "FLAMMABLE_ASH", "NOCOLLIDE", "FLOWER" ],
+    "examine_action": "harvest_furn",
+    "harvest_by_season": [ { "seasons": [ "spring", "summer", "autumn" ], "id": "olieribos_harv" } ],
+    "bash": { "str_min": 2, "str_max": 6, "sound": "crunch.", "sound_fail": "whish." }
+  },
+  {
+    "type": "furniture",
+    "id": "f_glowing_mushroom",
+    "name": "glowing mushrooms",
```
Comment (Night-Pryanik) [inline data/mods/Magiclysm/furniture_and_terrain/furniture_plants.json:65]: Plural name needed.

File: data/mods/Magiclysm/furniture_and_terrain/furniture_plants.json [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+    "flags": [ "FLAMMABLE_ASH", "ORGANIC", "REDUCE_SCENT" ],
+    "examine_action": "harvest_furn",
+    "harvest_by_season": [ { "seasons": [ "summer", "autumn" ], "id": "sojourners_palm_harv" } ],
+    "bash": { "str_min": 2, "str_max": 6, "sound": "crunch.", "sound_fail": "whish." }
+  },
+  {
+    "type": "furniture",
+    "id": "f_whiskflower",
+    "name": "Whiskflower bush",
+    "description": "A small plant with curled leaves and  pink whisk-shaped flowers.  It has a very strong alcoholic smell.",
```
Comment (Night-Pryanik) [inline data/mods/Magiclysm/furniture_and_terrain/furniture_plants.json:200]:
```suggestion
    "description": "A small plant with curled leaves and pink whisk-shaped flowers.  It has a very strong alcoholic smell.",
```

File: data/mods/Magiclysm/items/herbs_and_flowers.json [hunk 14]
```diff
@@ -0,0 +1,220 @@
...
+    "weight": "2 g",
+    "volume": "3 ml",
+    "looks_like": "wheat"
+  },
+  {
+    "type": "COMESTIBLE",
+    "id": "glowing_mushroom",
+    "name": { "str_sp": "glowing mushroom caps" },
+    "//": "'light' related, partially 'speed'",
+    "description": "Slim stripes cut from a bioluminescent mushroom, still surrounded by a fain glowing aura.",
```
Comment (Night-Pryanik) [inline data/mods/Magiclysm/items/herbs_and_flowers.json:70]:
```suggestion
    "description": "Slim stripes cut from a bioluminescent mushroom, still surrounded by a faint glowing aura.",
```

File: data/mods/Magiclysm/items/herbs_and_flowers.json [hunk 15]
```diff
@@ -0,0 +1,220 @@
...
+    "use_action": { "type": "cast_spell", "spell_id": "glowing_mushroom_spell", "no_fail": true, "level": 0 },
+    "looks_like": "mushroom",
+    "vitamins": [ [ "redcells", 25 ], [ "vitC", 6 ] ]
+  },
+  {
+    "type": "COMESTIBLE",
+    "id": "horn_moss_capsule",
+    "name": { "str_sp": "horn moss capsules" },
+    "//": "'healing' related",
+    "description": "Pure white moss capsules, shaped like a goat's horn.  The fine spores contained within are a powerfull magical reagent, and can be consumed to recover small amounts of health.",
```
Comment (Night-Pryanik) [inline data/mods/Magiclysm/items/herbs_and_flowers.json:87]:
```suggestion
    "description": "Pure white moss capsules, shaped like a goat's horn.  The fine spores contained within are a powerful magical reagent, and can be consumed to recover small amounts of health.",
```

**Pull request merged**